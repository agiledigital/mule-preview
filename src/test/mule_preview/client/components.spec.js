import React from "react";
import renderer from "react-test-renderer";
import {
  MuleComponent,
  MuleContainer
} from "../../../../dist/mule_preview.client.components";
import {
  jsToClj,
  makeAtom,
  makeSet
} from "../../../../dist/mule_preview.client.test_utils";
import { readFileSync } from "fs";
import { join } from "path";

// Made up mapping data
const mappings = JSON.parse(
  readFileSync(join(__dirname, "__fixtures__/example-mappings.json"))
);

describe("when rendering a Mule component", () => {
  describe("when the component has mapping", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleComponent
            name="set-payload"
            description="Set some payload"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 1, position: 4 })}
            showing-atom={makeAtom(false)}
            mappings={jsToClj(mappings)}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when the component is unknown", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleComponent
            name="sjdsfjlsdf"
            description="Set some payload"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 1, position: 4 })}
            showing-atom={makeAtom(false)}
            mappings={jsToClj(mappings)}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("when rendering a Mule container", () => {
  describe("when it has no children", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleContainer
            name="flow"
            description="Do something"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 2, position: 3 })}
            showing-atom={makeAtom(false)}
            mappings={jsToClj(mappings)}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when it has children", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <MuleContainer
            name="flow"
            description="Do something"
            css-class="some-css-class"
            content-root="some-content-root"
            location={jsToClj({ line: 2, position: 3 })}
            showing-atom={makeAtom(false)}
            mappings={jsToClj(mappings)}
          >
            <MuleComponent
              name="set-payload"
              description="Set some payload"
              css-class="some-css-class"
              content-root="some-content-root"
              location={jsToClj({ line: 1, position: 4 })}
              showing-atom={makeAtom(false)}
              labels={makeSet(["edited"])}
              mappings={jsToClj(mappings)}
            />
            <MuleComponent
              name="message-properties-transformer"
              description="Transform something"
              css-class="some-css-class"
              content-root="some-content-root"
              location={jsToClj({ line: 4, position: 4 })}
              showing-atom={makeAtom(false)}
              labels={makeSet(["removed"])}
              mappings={jsToClj(mappings)}
            />
            <MuleComponent
              name="http:request"
              description="Make some request"
              css-class="some-css-class"
              content-root="some-content-root"
              location={jsToClj({ line: 7, position: 4 })}
              showing-atom={makeAtom(false)}
              labels={makeSet(["added"])}
              mappings={jsToClj(mappings)}
            />
          </MuleContainer>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
