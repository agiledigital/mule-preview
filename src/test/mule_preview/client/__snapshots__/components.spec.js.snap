// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering a Mule component when the component has mapping renders correctly 1`] = `
<div>
  <div
    className="component-container some-css-class"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="component set-payload"
    >
      <img
        className="category-frame"
        src="/img/icons/setters.large.png"
      />
      <img
        className="icon"
        src="/img/icons/set-payload.png"
      />
      <div
        className="label"
      >
        Set some payload
      </div>
    </div>
  </div>
</div>
`;

exports[`when rendering a Mule component when the component is unknown renders correctly 1`] = `
<div>
  <div
    className="component-container some-css-class"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="component sjdsfjlsdf"
    >
      <img
        className="category-frame"
        src="/img/icons/org.mule.tooling.ui.modules.core.miscellaneous.large.png"
      />
      <img
        className="icon"
        src="/img/icons/UnknownNode-48x32.png"
      />
      <div
        className="label"
      >
        Set some payload
      </div>
    </div>
  </div>
</div>
`;

exports[`when rendering a Mule container when it has children renders correctly 1`] = `
<div>
  <div
    className="container mule-flow some-css-class"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="container-title"
    >
      Do something
    </div>
    <div
      className="container-inner"
    >
      <div
        className="icon-container"
      >
        <img
          className="category-frame"
          src="/img/icons/org.mule.tooling.ui.modules.core.miscellaneous.large.png"
        />
      </div>
      <div
        className="container-children"
      >
        <div>
          <div
            className="component-container some-css-class"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              className="diff-icon"
              src="/img/is-not-equal-to.svg"
            />
            <div
              className="component set-payload"
            >
              <img
                className="category-frame"
                src="/img/icons/setters.large.png"
              />
              <img
                className="icon"
                src="/img/icons/set-payload.png"
              />
              <div
                className="label"
              >
                Set some payload
              </div>
            </div>
          </div>
        </div>
        <img
          className="flow-arrow"
          src="/img/arrow-right-2x.png"
        />
        <div>
          <div
            className="component-container some-css-class"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              className="diff-icon"
              src="/img/minus.svg"
            />
            <div
              className="component message-properties-transformer"
            >
              <img
                className="category-frame"
                src="/img/icons/transformers.large.png"
              />
              <img
                className="icon"
                src="/img/icons/message-properties-transformer.png"
              />
              <div
                className="label"
              >
                Transform something
              </div>
            </div>
          </div>
        </div>
        <img
          className="flow-arrow"
          src="/img/arrow-right-2x.png"
        />
        <div>
          <div
            className="component-container some-css-class"
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          >
            <img
              className="diff-icon"
              src="/img/plus.svg"
            />
            <div
              className="component http:request"
            >
              <img
                className="category-frame"
                src="/img/icons/listeners.large.png"
              />
              <img
                className="icon"
                src="/img/icons/http-request.png"
              />
              <div
                className="label"
              >
                Make some request
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`when rendering a Mule container when it has no children renders correctly 1`] = `
<div>
  <div
    className="container mule-flow some-css-class"
    onMouseOut={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="container-title"
    >
      Do something
    </div>
    <div
      className="container-inner"
    >
      <div
        className="icon-container"
      >
        <img
          className="category-frame"
          src="/img/icons/org.mule.tooling.ui.modules.core.miscellaneous.large.png"
        />
      </div>
      <div
        className="container-children"
      />
    </div>
  </div>
</div>
`;
